sktime/utils/validation/tests/test_write_dataframe_to_ts.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+__author__ = ["Jason Pong"]
+__all__ = ["test_write_dataframe_to_ts_success", "test_write_dataframe_to_ts_fail"]
+
+import os
+
+import numpy as np
+import pytest
+from pandas._testing import assert_frame_equal
+
+import sktime
+
+from sktime.utils.load_data import load_from_tsfile_to_dataframe
+from sktime.utils.results_writing import write_dataframe_to_tsfile
+
+
+def test_write_dataframe_to_ts_success(tmp_path):
+    # load an example dataset
+    path = os.path.join(
+        os.path.dirname(sktime.__file__),
+        "datasets/data/ItalyPowerDemand/ItalyPowerDemand_TEST.ts",
+    )
+    test_x, test_y = load_from_tsfile_to_dataframe(path)
+    # output the dataframe in a ts file
+    write_dataframe_to_tsfile(
+        data=test_x,
+        path=tmp_path,
+        problem_name="ItalyPowerDemand",
+        timestamp=False,
+        univariate=True,
+        class_label=[1, 2],
+        class_value_list=test_y,
+        comment="""
+          The data was derived from twelve monthly electrical power demand
+          time series from Italy and first used in the paper "Intelligent
+          Icons: Integrating Lite-Weight Data Mining and Visualization into
+          GUI Operating Systems". The classification task is to distinguish
+          days from Oct to March (inclusive) from April to September.
+        """,
+    )
+    # load data back from the ts file
+    result = f"{tmp_path}/ItalyPowerDemand/ItalyPowerDemand_transform.ts"
```

Comment:
Why is this called `ItalyPowerDemand_transform.ts`, where does the `transform` come from?

Reply:
the write_dataframe_to_tsfile appends _transformat the end of the problem name as the file name, see
https://github.com/whackteachers/sktime/blob/3591a48577a7cda6fb2b5fffa8cdf1f6597107c2/sktime/utils/results_writing.py#L194-L201

Reply:
hi, this is fine, but fyi general format for storing results we use is 
<problem name>/<problem_name>.ts
If I were doing a transform on multiple data I would either have a whole directory and keep the original name
<transform name>/<problem name>/<problem_name>.ts
or append the transform name to all
<problem name_transform name>/<problem_name_transform name>.ts

Reply:
ok thanks for the reminder, I will keep this in mind next time

Reply:
from a user perspective, I find it strange if I give a path (directory) and problem name but then can't find the generated file, no? would probably prefer to have a user friendly function which takes data and file (path with file name, e.g. "my_data.ts") and a wrapper for your benchmarking purposes

Reply:
one reason for this is that we need to organise multiple problems, each of which has 6 or more files associated with it. These may be predefined splits into train and test or resamples. This structure allows us to automate much of the internal data complexity without bothering the user. For example, if you load a problem it looks to detect if there is already an predefined train test split, and it also looks to see if the resamples are already generated.

Reply:
yes I think that makes sense, but can that not be handled by a wrapper which defines additional behaviour for your purposes, so that we can still have a more generic function for writing a data frame to a ts file for users, i.e. the counterpart to `load_from_tsfile_to_dataframe`

Reply:
Ultimately, sure, but that's beyond the scope of this PR